index.js

```diff
@@ -93,12 +95,36 @@ class Inspector {
         // This is a way to handle sending information about thread back
         if (pearInspectMethod === 'connect') {
           session = new this.inspector.Session()
+
+          const isBareInspector = !session.disconnect
```

Comment:
This will cause issues if `bare-inspector` adds a `disconnect()` method. I'd use `which-runtime` to check if this is running in Bare, in which case you can assume that the session will be from `bare-inspector`.

Reply:
The module might also very well be reaching a point where splitting the Node.js and Bare implementations using an `exports` map makes sense.